investment_tracking/views.py

```diff
@@ -18,14 +20,43 @@ class InvestmentTrackingSaveView(APIView):
     def post(self, request, startup_id):
         investor_profile = get_object_or_404(InvestorProfile, user=request.user)
         startup = get_object_or_404(StartUpProfile, id=startup_id)
+
         serializer = InvestmentTrackingSerializerCreate(data={'investor': investor_profile.id, 'startup': startup.id})
         if serializer.is_valid():
             try:
                 investment_tracking = InvestmentTracking(investor=investor_profile, startup=startup)
                 investment_tracking.save()
-                return Response({"message": "StartUp has been successfully saved."}, status = status.HTTP_201_CREATED)
+                return Response({"message": "StartUp has been successfully saved."}, status=status.HTTP_201_CREATED)
             except IntegrityError:
-                return Response({"message": "Startup is already saved."}, status = status.HTTP_400_BAD_REQUEST)
+                return Response({"message": "Startup is already saved."}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class InvestmentTrackingListView(generics.ListAPIView):
+    """
+    API view to get list all startups saved by the investor.
+    """
+    permission_classes = [IsAuthenticated]
+    queryset = InvestmentTracking.objects.all()
+    serializer_class = ListInvestmentTrackingSerializer
+    filter_backends = [DjangoFilterBackend, OrderingFilter]
+    filterset_fields = ['startup__name']
+    ordering_fields = ['saved_at', 'startup__name']
+
+    def get_queryset(self):
+        investor_profile = get_object_or_404(InvestorProfile, user=self.request.user)
+        return InvestmentTracking.objects.filter(investor=investor_profile)
+
+
+class InvestmentTrackingUnsaveView(APIView):
+    """
+    API view to unfollow a saved startup.
+    """
+    permission_classes = [IsAuthenticated]
+
+    def delete(self, request, startup_id):
+        investor_profile = get_object_or_404(InvestorProfile, user=request.user)
```

Comment:
Refactor Duplicate Code (DRY Principle)

Problem: Repeated get_object_or_404 calls for InvestorProfile in each view.

Solution: Create a helper method to retrieve InvestorProfile to streamline code and reduce redundancy.

def get_investor_profile(self, request):
    return get_object_or_404(InvestorProfile, user=request.user)

hen call self.get_investor_profile(request) in each view to retrieve the profile.

Reply:
Done